.github/workflows/eval-ui-tests-p0.yml

```diff
@@ -0,0 +1,104 @@
+name: Maestro UI Tests
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    paths:
+      - 'Sources/**'
+      - '.github/workflows/eval-ui-tests-p0.yml'
+      - 'fastlane/**'
+      - 'Shared/**'
+      - 'TestSamples/**'
+
+# https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  build-sample-for-maestro:
```

Comment:
`m`: I would prefer not to add maestro in the job and workflow name because we plan on keeping this no matter the underlying technology. If we decided to switch over to XCTest, we would have to rename this, which impacts the CI statistics in Sentry, for example.

Reply:
Make sense, I had the specific names to isolate the stats when testing.